database/secret.go

```diff
@@ -54,6 +54,10 @@ type Secret struct {
 	Images       pq.StringArray `sql:"images" gorm:"type:varchar(1000)"`
 	Events       pq.StringArray `sql:"events" gorm:"type:varchar(1000)"`
 	AllowCommand sql.NullBool   `sql:"allow_command"`
+	CreatedAt    sql.NullInt64  `sql:"created_at"`
+	CreatedBy    sql.NullInt64  `sql:"created_by"`
+	UpdatedAt    sql.NullInt64  `sql:"updated_at"`
+	UpdatedBy    sql.NullInt64  `sql:"updated_by"`
```

Comment:
Is it safe to assume that `CreatedBy` and `UpdatedBy` are `int64` types since they'll store the user's ID?

https://github.com/go-vela/types/blob/911ffbea2e427883ccca2cc3f3769fdae75e0946/database/user.go#L48-L58

IIRC, they were `string` types earlier which is why I'm asking.

Reply:
Yes after taking a look at that User struct, I thought it best to change those to int64 types. Originally I thought it might be stored as a name, but I believe int64 is best.

Reply:
@ecrupper ok thanks for the extra information 👍 

Out of curiosity, could you elaborate on why `int64` is the approach we're choosing?

I think I tend to agree that from the data storage perspective, `int64` is better than `string` for the database.

However, my concern with this approach has to do with data access and usage patterns.

Today, there isn't a great way to query a user by their `ID` in Vela.

So if I'm looking at a secret with the UI/CLI/API, I would only be able to see the user's `ID` who created/updated it.

To actually see what user created/updated the secret, I'd have to ask an admin "Who is user <id> in Vela?"

Perhaps one way to address these concerns is to add a new endpoint to Vela which allows looking a user up by their ID?

Otherwise we could consider storing the user's `Name` rather than `ID` in the `secret` object?

What are your thoughts on this?

Reply:
That's a great point. Originally I was thinking in terms of consistency (if there's a User table with IDs as their primary key, it would make sense to have any reference to a User  in another table be their ID). However, after considering the end user as you said, I agree that it should be a string. Not only would it save the UI an extra API call, but considering the CLI, running another command to find the name associated with a User ID seems tedious. I'll make that change!

Reply:
fwiw, i proposed user id initially to be consistent with the repo object in the `repos` table where we use `int64` for user id. another instance where user _name_ could be beneficial when trying to figure out who "owns" a repo.

in the future we might consider embedding the user struct in the response similar to how github does it for some of it's objects (eg. check response here: https://docs.github.com/en/rest/reference/issues#get-an-issue).

(waiting for @JordanSussman to chime in with the obligatory GraphQL commentary 😄)

thanks for the work and revisions @ecrupper !

Reply:
> (waiting for @JordanSussman to chime in with the obligatory GraphQL commentary 😄)

Don't tempt me a good time! Would love to add a v2 API with graphql and/or parameters to allow parsed output.

![graphql](https://pbs.twimg.com/media/DgsXLk_X4AEKiJJ?format=jpg&name=medium)